### PR TITLE
[Merge]SessionTokenAndDocumentProducerSupport

### PR DESCRIPTION
**Pull request by xinlian12**
**Description:**
Part of the effort of adding Merge support in Java SDK. Main working item:https://github.com/Azure/azure-sdk-for-java/issues/32131

It imported the following two related fixes from .net:
https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2531
https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2980

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java [hunk 1]
```diff
@@ -226,24 +229,46 @@ private Flux<DocumentProducerFeedResponse> splitProof(Flux<DocumentProducerFeedR
...
+                            logger.error("Failed to find at least one child range");
+                            return Mono.error(new IllegalStateException("Failed to find at least one child range"));
+                        }
+
+                        if (partitionKeyRangesValueHolder.v.size() == 1) {
+                            // The feedRange is gone due to merge
+                            // we are going to continue drain the current document producer
+                            // Due to the feedRange does not cover full partition anymore, during populateHeaders, startEpk and endEpk headers will be added
+                            if (logger.isDebugEnabled()) {
+                                logger.info(
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java:244]: `logger.debug` or `logger.isInfoEnabled()` 
We should make this consistent with the below logging as well.

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @xinlian12

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java:244]: Good catch, updated

**Comment by xinlian12** [comment]
The test failed is conflictCustomLWW, which is a known flaky tests. 
This PR does not touch any logic related to conflicts.

**Pull request merged**